web/wp-content/themes/borealis-2021/single-news.php

```diff
@@ -8,17 +8,25 @@
  * @package pg-wp-starter
  */
 
-get_header();
-$hero_image_url = get_the_post_thumbnail_url($post->ID, 'full');
-$title = get_the_title($post->ID);
-$publication_date = get_post_meta($post->ID, 'publication_date', true);
-$authors = get_post_meta($post->ID, 'authors', true);
-$url = get_permalink($post->ID);
+  get_header();
+  $thumbnail_id = get_post_thumbnail_id( $post->ID );
+  $alt = get_post_meta($thumbnail_id, '_wp_attachment_image_alt', true); 
+
+
+  $title = get_the_title($post->ID);
+  $publication_date = get_post_meta($post->ID, 'publication_date', true);
+  $authors = get_post_meta($post->ID, 'authors', true);
+  $url = get_permalink($post->ID);
+
 ?>
-<?php if (isset($hero_image_url) && !empty($hero_image_url)) : ?>
-    <div aria-hidden="true" class="pt-100 mt-19 bg-cover bg-no-repeat bg-center" style="background-image: url(<?php echo esc_url_raw($hero_image_url) ?>)"></div>
+<?php if (isset($thumbnail_id) && !empty($thumbnail_id)) : ?>
```

Comment:
```suggestion
<?php if (!empty($thumbnail_id)) : ?>
```

you do not need `isset` as that will only fail if the variable is missing or the value is `null` ([source](https://www.php.net/manual/en/function.isset.php)) 

You set the value of `$thumbnail_id` to the return val of `get_post_thumbnail_id` on line `12`.  That function:

https://developer.wordpress.org/reference/functions/get_post_thumbnail_id/

Will return:

`(int|false) Post thumbnail ID (which can be 0 if the thumbnail is not set), or false if the post does not exist.`

so you will NEVER have a value for `isset($thumbnail_id)` that is ever anything other than `true`.

Reply:
Great thank you that's excellent to know. Will watch out for same pattern in future.